test/performance/scale_revision_by_load_test.go

```diff
@@ -0,0 +1,202 @@
+	}
+	clients := perfClients.E2EClients
+
+	defer TearDown(perfClients, names, t.Logf)
+	test.CleanupOnInterrupt(func() { TearDown(perfClients, names, t.Logf) })
+
+	t.Log("Creating a new Service")
+	objs, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{
+		ContainerResources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			},
+		}},
+		testingv1alpha1.WithConfigAnnotations(map[string]string{"autoscaling.knative.dev/target": fmt.Sprintf("%d", targetConcurrency)}),
+	)
+	if err != nil {
+		t.Fatalf("Failed to create Service: %v", err)
+	}
+
+	domain := objs.Route.Status.Domain
+	endpoint, err := ingress.GetIngressEndpoint(clients.KubeClient.Kube)
+	if err != nil {
+		t.Fatalf("Cannot get service endpoint: %v", err)
+	}
+
+	// Make sure we are ready to serve.
+	st := time.Now()
+	t.Log("Starting to probe the endpoint at", st)
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		domain+"/?timeout=10", // To generate any kind of a valid response.
+		test.RetryingRouteInconsistency(func(resp *spoof.Response) (bool, error) {
+			return resp.StatusCode == 200, nil
+		}),
+		"WaitForEndpointToServeText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint at domain %s didn't serve the expected response: %v", domain, err)
+	}
+	t.Logf("Took %v for the endpoint to start serving", time.Since(st))
+
+	scaleChannel := make(chan *scaleEvent, 100)
+	stopInformer := make(chan struct{})
+
+	factory := informers.NewSharedInformerFactory(clients.KubeClient.Kube, 0)
+	endpointsInformer := factory.Core().V1().Endpoints().Informer()
+	endpointsInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			newEndpoints := newObj.(*corev1.Endpoints)
+			if strings.Contains(newEndpoints.GetName(), names.Service) {
+				newNumAddresses := resources.ReadyAddressCount(newEndpoints)
+				oldNumAddresses := resources.ReadyAddressCount(oldObj.(*corev1.Endpoints))
+				if newNumAddresses != oldNumAddresses {
+					event := &scaleEvent{
+						oldScale:  oldNumAddresses,
+						newScale:  newNumAddresses,
+						timestamp: time.Now(),
+					}
+					scaleChannel <- event
+				}
+			}
+		},
+	})
+
+	go func() {
+		defer close(scaleChannel)
+		endpointsInformer.Run(stopInformer)
+	}()
+
+	opts := loadgenerator.GeneratorOptions{
+		Duration:       iterationDuration,
+		NumThreads:     numClients,
+		NumConnections: numClients,
+		Domain:         domain,
+		QPS:            qpsPerClient * float64(numClients),
+		URL:            fmt.Sprintf("http://%s/?timeout=%d", *endpoint, processingTimeMillis),
+	}
+
+	t.Logf("Starting test with %d clients at %s", numClients, time.Now())
+	resp, err := opts.RunLoadTest(false)
+	if err != nil {
+		t.Fatalf("Generating traffic via fortio failed: %v", err)
+	}
+
+	close(stopInformer)
```

Comment:
If you use the helper I mentioned above, you can `close(scaleChannel)` at this point as well, however you'll need to adjust the event handler in the informer slightly to:

```golang
select {
  case scaleChannel <- event:
  default:
}
```

To make sure you're no longer writing on a closed channel.